src/components/GoogleAuth.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { googleSignIn, googleSignOut } from '../actions';
+import { GOOGLE_AUTH_PUBLIC } from '../apis/constants';
+
+class GoogleAuth extends React.Component {
+    componentDidMount() {
+        window.gapi.load('client:auth2', () => {
+          window.gapi.client
+            .init({
+              clientId: GOOGLE_AUTH_PUBLIC,
+              scope: 'email'
+            })
+            .then(() => {
+              this.auth = window.gapi.auth2.getAuthInstance();
+    
+              this.onAuthChange(this.auth.isSignedIn.get());
+              this.auth.isSignedIn.listen(this.onAuthChange);
+            });
+        });
+    }
+
+    onAuthChange = isSignedIn => {
+        if (isSignedIn) {
+            this.props.googleSignIn(this.auth.currentUser.get().getId());
+        } else {
+            this.props.googleSignOut();
+        }
+    };
+
+    onSignInClick = () => {
+        this.auth.signIn();
+    };
+
+    onSignOutClick = () => {
+        this.auth.signOut();
+    };
+
+    renderAuthButton() {
+        if (this.props.isSignedIn) {
+            return (
+            <button onClick={this.onSignOutClick} className="ui red google button">
+                <i className="google icon" />
+                Sign Out
+            </button>
+            );
+        } else {
+            return (
+            <button onClick={this.onSignInClick} className="ui red google button">
+                <i className="google icon" />
+                Sign In with Google
+            </button>
+            );
+        }
+    }
+
+    render() {
+        return (
+            <div>
+                {this.renderAuthButton()}
+            </div>
+        );
+    }
+}
+const mapStateToProps = (state) => {
```

Comment:
you can just do the selection here if you want: `mapStateToProps = ({googleAuth: {isSignedIn}) => ({isSignedIn})`. Since you are only really selecting one portion of the state. As the need for more properties expands then you could think about using `reselect` library.